src/mesh/boundary_standard.cxx

```diff
@@ -23,88 +23,92 @@
     lead to an out of bounds access error later but we add it here to provide a
     more explanatory message.
  */
-void verifyNumPoints(BoundaryRegion *region, int ptsRequired) {
-  TRACE("Verifying number of points available for BC");
+namespace {
```

Comment:
Are all the changes in this file separate from the FieldData gaining a mesh pointer?

Reply:
I guess the changes are separate in the sense that they could all be made without adding a mesh pointer to FieldData, but the mesh pointer in FieldData is required for the boundary conditions to work correctly when the fieldmesh is not the same as the global mesh.
* in verifyNumPoints we need to get localmesh from the `BoundaryRegion *region`; `BoundaryRegion*`s are created in a method of `FieldData`; `FieldData` had to gain a mesh pointer so that `BoundaryRegion*`s aren't always created with the global `mesh`.
* the `BoundaryRegion* bndry` of the `BoundaryOp`s could previously be wrong if the global and local `Mesh` are different (e.g. wrong number of points, or wrong number of guard cells, etc.).